dimod/core/bqm.py

```diff
@@ -727,6 +727,30 @@ def min_and_max(iterable):
                        ignored_interactions=ignored_interactions,
                        ignore_offset=ignore_offset)
 
+    def relabel_variables_as_integers(self, inplace=True):
+        """Relabel the variables of the BQM to integers.
+
+        Args:
+            inplace (bool, optional, default=True):
+                If True, the binary quadratic model is updated in-place;
+                otherwise, a new binary quadratic model is returned.
+
+        Returns:
+            tuple: A 2-tuple containing:
+
+                A binary quadratic model with the variables relabeled. If
+                `inplace` is set to True, returns itself.
+
+                dict: The mapping that will restore the original labels.
+
+        """
+        if not inplace:
+            return self.copy().relabel_variables(inplace=True)
+
+        mapping = dict((v, i) for i, v in enumerate(self.variables) if i != v)
```

Comment:
Is there a cost to relabeling variables to the same label?

Reply:
Not really, but cleaner this way